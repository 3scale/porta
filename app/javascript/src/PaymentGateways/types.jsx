// @flow

export type BraintreeFormProps = {
  // eslint-disable-next-line flowtype/no-weak-types
  braintreeClient: any,
  billingAddress?: string,
  threeDSecureEnabled: boolean,
  formActionPath: string,
  countriesList: string,
  selectedCountryCode?: string
}
