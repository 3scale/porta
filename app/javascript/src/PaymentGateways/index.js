// @flow

export { StripeFormWrapper } from './stripe/components/StripeFormWrapper'
export { StripeCardForm } from './stripe/components/StripeCardForm'
export { BraintreeForm } from './braintree/BraintreeForm'
export { BraintreeFormWrapper } from './braintree/BraintreeFormWrapper'
export { BraintreeCardFields } from './braintree/BraintreeCardFields'
export { BraintreeBillingAddressFields } from './braintree/BraintreeBillingAddressFields'
export { BraintreeUserFields } from './braintree/BraintreeUserFields'
export { BraintreeSubmitFields } from './braintree/BraintreeSubmitFields'
export * from './braintree/braintree'
export { Label } from './braintree/components/Label'
export { Input } from './braintree/components/Input'
export { ListItem } from './braintree/components/ListItem'
export * from './braintree/types'
