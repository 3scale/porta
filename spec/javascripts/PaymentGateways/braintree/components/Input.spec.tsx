import React from 'react'
import { mount } from 'enzyme'
import { Input } from 'PaymentGateways'

const onChangeSPy = jest.fn()
const props = {
  id: 'my-id',
  required: true,
  name: 'username',
  value: 'Shaquille O Neal',
  onChange: onChangeSPy
}

it('should render properly', () => {
  const wrapper = mount(<Input {...props} />)
  expect(wrapper.exists()).toBe(true)
})

it('should render with right props', () => {
  const wrapper = mount(<Input {...props} />)
  expect(wrapper.props().id).toEqual('my-id')
  expect(wrapper.props().required).toEqual(true)
  expect(wrapper.props().name).toEqual('username')
  expect(wrapper.props().value).toEqual('Shaquille O Neal')
})

it('should call onChange method', () => {
  const event = { currentTarget: { value: 'Magic Johnson' } }
  const wrapper = mount(<Input {...props} />)
  wrapper.props().onChange(event)
  expect(onChangeSPy).toHaveBeenCalledWith(event)
})
