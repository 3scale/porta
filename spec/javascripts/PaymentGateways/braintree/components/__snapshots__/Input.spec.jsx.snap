// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<Input
  id="my-id"
  name="username"
  onChange={[MockFunction]}
  required={true}
  value="Shaquille O Neal"
>
  <input
    className="col-md-6 form-control"
    id="my-id"
    name="username"
    onChange={[MockFunction]}
    required={true}
    type="text"
    value="Shaquille O Neal"
  />
</Input>
`;
